packages/api/lib/rulesHelpers.js

```diff
@@ -261,7 +268,407 @@ async function deleteRuleResources(knex, rule) {
+        }
+      });
+    }
+    /* eslint-enable no-loop-func */
+    if (subExists) break;
+  }
+  while (token);
+  /* eslint-enable no-await-in-loop */
+  if (!subExists) {
+    // create sns subscription
+    const subscriptionParams = {
+      TopicArn: item.value,
+      Protocol: 'lambda',
+      Endpoint: process.env.messageConsumer,
+      ReturnSubscriptionArn: true,
+    };
+    const r = await awsServices.sns().subscribe(subscriptionParams).promise();
+    subscriptionArn = r.SubscriptionArn;
+    // create permission to invoke lambda
+    const permissionParams = {
+      Action: 'lambda:InvokeFunction',
+      FunctionName: process.env.messageConsumer,
+      Principal: 'sns.amazonaws.com',
+      SourceArn: item.value,
+      StatementId: `${item.name}Permission`,
+    };
+    await awsServices.lambda().addPermission(permissionParams).promise();
+  }
+  return subscriptionArn;
+}
+
+/**
+ * Update the event source mappings for Kinesis type rules.
+ *
+ * Avoids object mutation by cloning the original rule item.
+ *
+ * @param {PostgresRule} ruleItem
+ *   A rule item
+ * @param {Object} ruleArns
+ *   An object containing arn and logEventArn values
+ * @param {string} ruleArns.arn
+ *   UUID for event source mapping from Kinesis stream for messageConsumer Lambda
+ * @param {string} ruleArns.logEventArn
+ *   UUID for event source mapping from Kinesis stream to KinesisInboundEventLogger Lambda
+ * @returns {PostgresRule}
+ *   Updated rule item
+ */
+function updateKinesisRuleArns(ruleItem, ruleArns) {
+  const updatedRuleItem = cloneDeep(ruleItem);
+  updatedRuleItem.arn = ruleArns.arn;
+  updatedRuleItem.log_event_arn = ruleArns.logEventArn;
+  return updatedRuleItem;
+}
+
+/**
+   * Adds CloudWatch event rule and target
+   *
+   * @param {PostgresRule} item - The rule item
+   * @param {Object} payload    - The payload input of the CloudWatch event
+   * @returns {void}
+   */
+async function addRule(item, payload) {
+  const name = `${process.env.stackName}-custom-${item.name}`;
+  const state = item.enabled ? 'ENABLED' : 'DISABLED';
+  await CloudwatchEvents.putEvent(
+    name,
+    item.value,
+    state,
+    'Rule created by cumulus-api'
+  );
+  const targetId = 'lambdaTarget';
+
+  await CloudwatchEvents.putTarget(
+    name,
+    targetId,
+    process.env.invokeArn,
+    JSON.stringify(payload)
+  );
+}
+
+/*
+ * Checks if record is valid
+ *
+ * @param {PostgresRule} rule - Rule to check validation
+ * @returns {void}            - Returns if record is valid, throws error otherwise
+ */
+function recordIsValid(rule) {
+  const error = new Error('The record has validation errors');
+  error.name = 'SchemaValidationError';
+  if (!rule.name) {
+    error.detail = 'Rule name is undefined.';
+    throw error;
+  }
+  if (!rule.workflow) {
+    error.detail = 'Rule workflow is undefined.';
+    throw error;
+  }
+  if (!rule.type) {
+    error.detail = 'Rule type is undefined.';
+    throw error;
+  }
```

Comment:
`createRuleTrigger` sets the default value for `enabled` to be true if it is undefined. Did you still want me to add that extra condition? It will almost certainly always be set. 🤔

Reply:
Did we add these unit tests for the endpoint?